wordpress-mobile/WordPress-Android
# Update release-toolkit to 1.1.0, enable update check, and re-enable TypographyDashes lint rule

### What

* Updates the release-toolkit to the latest 1.1.0
* Uses its new `check_for_toolkit_updates` action to be sure that we always pull the latest bugfixes from the toolkit
* Re-enable the `TypographyDashes` lint rule as error – to revert the changes made in de6c19cf45c3658aa2d6262f63652173ba5a81ae / #14740 – now that the toolkit 1.1.0 is supposed to handle the substitutions for us.
* Update translations so that they got post-processed again but with the fix we made in toolkit 1.1.0 about en-dashes

_Note: targeting `release/17.5` so that this fix is already applied during any additional betas we might want to do during 17.5 consolidation cycle as well as for the final release of 17.5_

### To Test

* Run `bundle exec fastlane download_translations` and ensure that:
  * It checks for toolkit updates and says `"Your release toolkit is up-to-date! ✅"`
  * The linter passes without any error related to en-dash

---

Side note: `./gradlew lintWordPressVanillaRelease` fails on my machine and finds 4 errors `MissingSuperCall`. Not sure why since I haven't changed anything about this rule.
<details><summary>Output of the linter when run on my Mac</summary>

```
Lint found 4 errors, 307 warnings (24 errors, 27 warnings filtered by baseline lint-baseline.xml)
…
> Task :WordPress:lintWordpressVanillaRelease FAILED
…
> Lint found errors in the project; aborting build.
  
  Fix the issues identified by lint, or add the following to your build script to proceed with errors:
  ...
  android {
      lintOptions {
          abortOnError false
      }
  }
  ...
  
  The first 3 errors (out of 4) were:
  /Users/olivier/Documents/Dev/apps/WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/media/MediaBrowserActivity.java:556: Error: Overriding method should call super.onRequestPermissionsResult [MissingSuperCall]
      public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] results) {
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
  /Users/olivier/Documents/Dev/apps/WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/media/MediaSettingsActivity.java:841: Error: Overriding method should call super.onRequestPermissionsResult [MissingSuperCall]
      public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions,
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
  /Users/olivier/Documents/Dev/apps/WordPress-Android/WordPress/src/main/java/org/wordpress/android/ui/ShareIntentReceiverActivity.java:152: Error: Overriding method should call super.onRequestPermissionsResult [MissingSuperCall]
      public void onRequestPermissionsResult(int requestCode,
                  ~~~~~~~~~~~~~~~~~~~~~~~~~~
  
```
</details>
Unsure what's happening, since if they are genuine linter errors, then how did the lint just pass for #14779 which is also cut from the same `release/17.5` branch? But I'm heading out for my weekend, so I figured we'll let CI do the lint see how it goes.
